kernel/base/src/main/java/com/twosigma/beakerx/kernel/commands/MagicCommandFinder.java

```diff
@@ -55,22 +58,27 @@ public static MagicCommandFinder find(Code code, Map<String, MagicCommandFunctio
       if (functionality.isPresent()) {
         functionalityToRun.put(command, functionality.get());
       } else {
-        errors.add(processUnknownCommand(command, message, executionCount, messageCreator));
+        errors.add(processIllegalCommand("Cell magic " + command + " not found", message, executionCount, messageCreator));
       }
     });
     return new MagicCommandFinder(functionalityToRun, errors);
   }
 
+  private static boolean isCellmagicHeadNonEmpty(String command) {
+    List<String> tokens = new StrTokenizer(command).getTokenList();
+
+    return !(command.replace(tokens.get(0), "").replace(" ", "").length() < 1);
```

Comment:
this is too complicated and it is not needed.
when it comes time to process the body, check if it is empty and then raise the error.